File: Data Structures & Algorithms/subsets/submission-0.cpp
class Solution {
public:
    vector<vector<int>> ans;
    vector<int> subset;
    void backtrack(int i,vector<int>& nums){
        if (i>=nums.size()){
            ans.emplace_back(subset);
            return;
        }
        subset.push_back(nums[i]);
        backtrack(i+1,nums);
        subset.pop_back();
        backtrack(i+1,nums);

    }
    vector<vector<int>> subsets(vector<int>& nums) {
        if (nums.size()==0){
            ans.push_back({});
            return ans;
        }
        backtrack(0,nums);
        return ans;
    }
};

File: Data Structures & Algorithms/contains-duplicate-ii/submission-0.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        i=0
        j=1
        while i<=len(nums)-1:
            while j<=len(nums)-1:
                if i!=j and nums[i]==nums[j] and abs(i-j)<=k:
                    return True
                j+=1
            j=0
            i+=1
        return False

